### PR TITLE
Cache default enabled config

### PR DESCRIPTION
**Pull request by trask**
this was made possible by #1643

### CODE REVIEW

**Review by anuraaga — APPROVED** [review]

File: javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/extension/instrumentation/InstrumentationModule.java [hunk 1]
```diff
@@ -228,8 +229,17 @@ public int order() {
...
+
+  // visible for testing
+  static boolean initDefaultEnabled(Config config) {
+    return config.getBooleanProperty("otel.instrumentation.common.default-enabled", true);
+  }
+
+  // visible for testing
+  static boolean initEnabled(
+      Config config, List<String> instrumentationNames, boolean defaultEnabled) {
+    return config.isInstrumentationEnabled(instrumentationNames, defaultEnabled);
```
Comment (iNikem) [inline javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/extension/instrumentation/InstrumentationModule.java:243]: I don't understand why these methods a called `init`? They don't initialise anything. I am confused by them

File: javaagent-tooling/src/test/groovy/io/opentelemetry/javaagent/extension/instrumentation/InstrumentationModuleTest.groovy [hunk 2]
```diff
@@ -77,18 +77,12 @@ class InstrumentationModuleTest extends Specification {
...
-    def target = new TestInstrumentationModule(["test"])
-    target.extend(new AgentBuilder.Default())
 
     expect:
-    target.enabled == enabled
-    target.applyCalled == enabled
-
-    cleanup:
-    Config.INSTANCE = null
+    InstrumentationModule.initDefaultEnabled(config) == enabled
```
Comment (iNikem) [inline javaagent-tooling/src/test/groovy/io/opentelemetry/javaagent/extension/instrumentation/InstrumentationModuleTest.groovy:85]: Is this method called only from test? They I don't see any point in neither the method nor the test.

File: javaagent-tooling/src/test/groovy/io/opentelemetry/javaagent/extension/instrumentation/InstrumentationModuleTest.groovy [hunk 3]
```diff
@@ -99,20 +93,12 @@ class InstrumentationModuleTest extends Specification {
...
-    def target = new TestInstrumentationModule([name, altName])
-    target.extend(new AgentBuilder.Default())
-
     expect:
-    target.enabled == enabled
-    target.applyCalled == enabled
-
-    cleanup:
-    Config.INSTANCE = null
+    InstrumentationModule.initEnabled(config, [name, altName], false) == enabled
```
Comment (iNikem) [inline javaagent-tooling/src/test/groovy/io/opentelemetry/javaagent/extension/instrumentation/InstrumentationModuleTest.groovy:101]: This also does not look like a test for `InstrumentationModule`, more like for `Config`.

**Comment by trask** [comment]
@iNikem I inlined those methods and removed the corresponding tests. Previously I was trying (too hard?) to keep the tests intact, which was the purpose of extracting those methods. PTAL

File: javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/extension/instrumentation/InstrumentationModule.java [hunk 4]
```diff
@@ -72,7 +75,7 @@ public InstrumentationModule(List<String> instrumentationNames) {
       throw new IllegalArgumentException("InstrumentationModules must be named");
     }
     this.instrumentationNames = new LinkedHashSet<>(instrumentationNames);
-    enabled = Config.get().isInstrumentationEnabled(this.instrumentationNames, defaultEnabled());
+    enabled = Config.get().isInstrumentationEnabled(instrumentationNames, defaultEnabled());
```
Comment (iNikem) [inline javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/extension/instrumentation/InstrumentationModule.java:78]:
```suggestion
    this.enabled = Config.get().isInstrumentationEnabled(instrumentationNames, defaultEnabled());
```

**Review by iNikem — APPROVED** [review]

**Pull request merged**